### PR TITLE
fix: issue 232: extension should work with hard coded lake path in the lean4/lakePath setting

### LINKED ISSUE #232: `checkLakeVersion` breaks when used with a Lake not in `PATH`
When determining whether to use Lake when no toolchain file is present, `checkLakeVersion` is called:

https://github.com/leanprover/vscode-lean4/blob/a154c97ef6541ec7d1b6b5f7480d684c627d54c0/vscode-lean4/src/leanclient.ts#L168

`checkLakeVersion` invokes Lake via `batchExecute`:

https://github.com/leanprover/vscode-lean4/blob/a154c97ef6541ec7d1b6b5f7480d684c627d54c0/vscode-lean4/src/leanclient.ts#L546

which requires that the executable specified be in `PATH`:

https://github.com/leanprover/vscode-lean4/blob/a154c97ef6541ec7d1b6b5f7480d684c627d54c0/vscode-lean4/src/utils/batch.ts#L25-L29

Thus, the invocation fails when the Lake has been manually specified -- either through a general toolchain path or a direct Lake path.

Based on the comment in `batchExecute`, the `PATH` check only appears to be for testing purposes, so one way to fix this issue would be to condition the check based on whether `batchExecute` is being used for testing. Alternatively, a different method could be used for invoking Lake.

### PR DESCRIPTION
**Pull request by lovettchris**
fixes #232

### CODE REVIEW

**Review by Vtec234 — COMMENTED** [review]
Afaik any kind of "check if running in a test" outside of test code is an antipattern because at that point you are no longer testing the real program, just some test-specific version of it. But I'm not too familiar with what is happening here, so if you think this is necessary then LGTM.

**Comment by lovettchris** [comment]
> Afaik any kind of "check if running in a test" outside of test code is an antipattern because at that point you are no longer testing the real program, just some test-specific version of it. But I'm not too familiar with what is happening here, so if you think this is necessary then LGTM.

Agreed, I'm trying to keep it to a minimum.  The problem is mocha is changing the runtime behavior in unexpected ways which is what this code is dealing with.

File: vscode-lean4/src/utils/batch.ts [hunk 1]
```diff
@@ -16,16 +16,18 @@ export async function batchExecute(
...
-        // check if the command exists so as not to trigger that exception.
-
         try {
-            const fullPath = findProgramInPath(executablePath);
-            if (!fullPath) {
-                resolve(undefined);
-                return;
+            if (isRunningTest())
+            {
+                // The mocha test framework listens to process.on('uncaughtException')
```
Comment (Vtec234) [inline vscode-lean4/src/utils/batch.ts:22]: I understand now. So the question is, could we turn this behaviour off in Mocha and delete the whole `findProgramInPath` check? If not, let's go with the `isRunningTest` check.

Comment (Vtec234) [reply vscode-lean4/src/utils/batch.ts:22]: Hm, but why is the exception raised by `spawn` uncaught? Maybe that's the problem in the first place?

Comment (lovettchris) [reply vscode-lean4/src/utils/batch.ts:22]: I don't think so, the mocha code that registers `uncaughtException` is buried pretty deep in mocha, and I couldn't find a way to disable it or undo it.

Comment (lovettchris) [reply vscode-lean4/src/utils/batch.ts:22]: > Hm, but why is the exception raised by `spawn` uncaught? Maybe that's the problem in the first place?

yeah, it's weird I know, we have a try/catch which catches it, but spawn still raises it as a process.uncaughtException for some reason, and this is then causing mocha to fail the test.  It is all very weird.

Comment (lovettchris) [reply vscode-lean4/src/utils/batch.ts:22]: PS: this code was already there by the way, I'm just making it test only code now.

**Pull request merged**